service/worker/scheduler/spec.go

```diff
@@ -295,20 +295,20 @@ func (cs *CompiledSpec) getNextTime(jitterSeed string, after time.Time) getNextT
 		after = cs.spec.StartTime.AsTime().Add(-time.Second)
 	}
 
+	isExcluded := func(t time.Time) bool {
+		return t.IsZero() || cs.excluded(t)
+	}
+	pastEndTime := func(t time.Time) bool {
+		return cs.spec.EndTime != nil && t.After(cs.spec.EndTime.AsTime())
+	}
 	var nominal time.Time
-	for {
+	for isExcluded(nominal) {
```

Comment:
slightly tricky with that IsZero inside isExcluded... why not just write
```
for nominal.IsZero() || cs.excluded(nominal) {
```
?
the extra function is a little confusing since the zero time isn't "excluded", it's used as a sentinel value here